modules/videoio/src/cap_avfoundation_mac.mm

```diff
@@ -293,7 +293,7 @@ - (IplImage*)getOutput;
 
     [mCaptureSession release];
     [mCaptureDeviceInput release];
-    [mCaptureDevice release];
+    // [mCaptureDevice release]; fix #7833
```

Comment:
sure, will do, no problem. The issue existed on the latest commit from the master  - java was crashing after VideoCapture.release(). With this fix all works like a charm - tried to switch on/switch off the camera many times and no issues at all. @jbyu is awesome, I owe him some beer :)